src/cpp/src/sampler.hpp

```diff
@@ -375,7 +376,7 @@ GroupBeamSearcher::GroupBeamSearcher(SequenceGroup::Ptr sequence_group)
         m_parameters{m_sequence_group->get_sampling_parameters()},
         m_groups{m_parameters.num_beam_groups} {
     OPENVINO_ASSERT(m_sequence_group->num_running_seqs() == 1);
-    OPENVINO_ASSERT(m_parameters.num_beams % m_parameters.num_beam_groups == 0,
+    assert(m_parameters.num_beams % m_parameters.num_beam_groups == 0 &&
         "number of beams should be divisible by number of groups");
```

Comment:
why is it not a part of validation of generation config?

Reply:
should we clean up such places then? Because we have such check in validation of generation config

Reply:
No. The next line assumes the numbers are divisible `m_parameters.num_beams / m_parameters.num_beam_groups`. Assert makes this assumption explicit.

Implementations usually lack extra checks so I'd like to keep it for debug build.